exchanges/bybit/bybit.go

```diff
@@ -8,979 +8,2605 @@ import (
 	}
-
-	if params == nil && jsonPayload == nil {
-		params = url.Values{}
+	value := reflect.ValueOf(result)
+	if value.Kind() != reflect.Ptr {
+		return fmt.Errorf("expected a pointer, got %T", value)
 	}
-
-	if jsonPayload != nil {
-		jsonPayload["recvWindow"] = defaultRecvWindow
-	} else if params.Get("recvWindow") == "" {
-		params.Set("recvWindow", defaultRecvWindow)
+	response := &RestResponse{
+		Result: result,
 	}
-
 	err = by.SendPayload(ctx, f, func() (*request.Item, error) {
-		var (
-			payload       []byte
-			hmacSignedStr string
-			headers       = make(map[string]string)
-		)
-
-		if jsonPayload != nil {
-			headers["Content-Type"] = "application/json"
-			jsonPayload["timestamp"] = strconv.FormatInt(time.Now().UnixMilli(), 10)
-			jsonPayload["api_key"] = creds.Key
-			hmacSignedStr, err = getJSONRequestSignature(jsonPayload, creds.Secret)
-			if err != nil {
-				return nil, err
-			}
-			jsonPayload["sign"] = hmacSignedStr
-			payload, err = json.Marshal(jsonPayload)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
-			params.Set("api_key", creds.Key)
-			hmacSignedStr, err = getSign(params.Encode(), creds.Secret)
-			if err != nil {
-				return nil, err
-			}
-			headers["Content-Type"] = "application/x-www-form-urlencoded"
-			switch method {
-			case http.MethodPost:
-				params.Set("sign", hmacSignedStr)
-				payload = []byte(params.Encode())
-			default:
-				path = common.EncodeURLValues(path, params)
-				path += "&sign=" + hmacSignedStr
-			}
-		}
-
 		return &request.Item{
-			Method:        method,
-			Path:          endpointPath + path,
-			Headers:       headers,
-			Body:          bytes.NewBuffer(payload),
-			Result:        &result,
+			Method:        http.MethodGet,
+			Path:          endpointPath + bybitAPIVersion + path,
+			Result:        response,
 			Verbose:       by.Verbose,
 			HTTPDebugging: by.HTTPDebugging,
 			HTTPRecording: by.HTTPRecording}, nil
-	}, request.AuthenticatedRequest)
+	}, request.UnauthenticatedRequest)
 	if err != nil {
 		return err
 	}
-	return result.GetError(true)
+	if response.RetCode != 0 && response.RetMsg != "" {
+		return fmt.Errorf("code: %d message: %s", response.RetCode, response.RetMsg)
+	}
+	return nil
 }
 
 // SendAuthHTTPRequestV5 sends an authenticated HTTP request
-func (by *Bybit) SendAuthHTTPRequestV5(ctx context.Context, ePath exchange.URL, method, path string, params url.Values, result UnmarshalTo, f request.EndpointLimit) error {
+func (by *Bybit) SendAuthHTTPRequestV5(ctx context.Context, ePath exchange.URL, method, path string, params url.Values, arg, result interface{}, f request.EndpointLimit) error {
+	val := reflect.ValueOf(arg)
+	if arg != nil && val.Kind() != reflect.Ptr {
+		return errNonePointerArgument
+	}
+	val = reflect.ValueOf(result)
+	if val.Kind() != reflect.Ptr {
+		return errNonePointerArgument
+	} else if val.IsNil() {
+		return errNilArgument
+	}
```

Comment:
But as the method is not returning the result I have to make sure the model is a pointer and is the response is marshaled into it.

Reply:
I've benchmarked it, and its only a 1.5ns check, I see you already did it for okx, so I think my concern about anything using `reflect` is slow is unfounded. I think it can stay, thank you for your explanation Sam